django/db/models/query.py

```diff
@@ -1523,20 +1524,43 @@ def prefetch_one_level(instances, prefetcher, lookup, level):
     Return the prefetched objects along with any additional prefetches that
     must be done due to prefetch_related lookups found from default managers.
     """
-    # prefetcher must have a method get_prefetch_queryset() which takes a list
-    # of instances, and returns a tuple:
 
-    # (queryset of instances of self.model that are related to passed in instances,
-    #  callable that gets value to be matched for returned instances,
-    #  callable that gets value to be matched for passed in instances,
-    #  boolean that is True for singly related objects,
-    #  cache name to assign to).
+    # On some databases (e.g. SQLite) the maximum number of variables in a query
+    # limits the maximum number of instances we can prefetch at once. In this case,
+    # do the query in batches.
+    connection = connections[instances[0]._state.db]
+    if connection.features.max_query_params is None:
+        batch_size = len(instances)
+    else:
+        # Subtract a safety margin because the queryset might contain custom filtering
+        # expressions that also use variables.
+        batch_size = connection.features.max_query_params - 25
```

Comment:
Would be fine I guess. I'm not too sure about the safety margin aspect in the first place, but it's SQLite so whatever works, I guess.